ninja/signature/utils.py

```diff
@@ -45,9 +55,36 @@ def make_forwardref(annotation: str, globalns: DictStrAny) -> Any:
     return evaluate_forwardref(forward_ref, globalns, globalns)
 
 
-def get_path_param_names(path: str) -> Set[str]:
-    "turns path string like /foo/{var}/path/{another}/end to set ['var', 'another']"
-    return {item.strip("{}") for item in re.findall("{[^}]*}", path)}
+def get_path_param_names_types(path: str) -> Dict[str, Union[type, None]]:
+    """turns path string like /foo/{var}/path/{int:another}/end to dict {'var': None, 'another': int}"""
+    names_types = (
+        ([None] + item.strip("{}").split(":"))[-1:-3:-1]
+        for item in re.findall("{[^}]*}", path)
+    )
+    return {name: _path_converter_type(type_) for name, type_ in names_types}
+
+
+def _path_converter_type(converter_name: str) -> Union[type, None]:
```

Comment:
Hey again, Stephen! I enjoy reading your code lines.

In these cases, is there a special explanation of `Union[T, None]` usage instead of `Optional[T]`?

Reply:
I am very new to typing, so my ignorance is the only explanation.